### ISSUE TITLE
Suggestion for improvement

### ISSUE THREAD

**Issue by michaldejmek** [joeyates/imap-backup#165]
First of all, I would like to thank you for the excellent app - it works great. 

Would it be possible to add the following features to the app? 

- create a log for each output that the application performs
- the ability to import accounts, for example using csv
- the possibility to start a backup using a command, not a json file

Thank you Michal.

**Comment by joeyates** [comment]
Hi @michaldejmek 

As far as logging goes, if you activate verbose logging, you should already get info about each action performed.

To import an account, the JSON configuration would need to be modified. What scenario are you imagining? Would you like to configure a very large number of accounts, or would this be done to avoid the menu-driven setup system?

Currently, you cannot run a backup directly from command-line parameters, but you **can** create a temporary JSON file and pass that as a parameter.

**Comment by michaldejmek** [comment]
Hi @joeyates 

Inserting an account is now a complicated process for more than 3 users.  Ideally I would like to import using a CSV file where I will have all the necessary information listed. 

Before I start backing up all users, it would be a good idea to check that all users can log in to the server.  Avoid the problem of typos .

**Comment by joeyates** [comment]
I think it would work to have a Ruby script to do this. The script could make use of the imap-backup library functions.

I imagine this being a non-core script, though. There are so many varied requirements, e.g. CSV formats and column naming are always such a problem. So, I see this as something to put in a `contrib` or `examples` directory so that users can fork it and adapt it to their needs.

The example script would need to reference imap-backup using [bundler/inline](https://bundler.io/guides/bundler_in_a_single_file_ruby_script.html), e.g.

```ruby
require "bundler/inline"

gemfile do
  source "https://rubygems.org"

  gem "imap-backup", ">=12.0.0"
end

...
```

It would then need to load the existing config file and merge in additions.

**Comment by joeyates** [comment]
I've added an example implementation of a [CSV import script in the `contrib` directory](https://github.com/joeyates/imap-backup/tree/a5c52badd6a9b9228f719d14d1a1704eb3c68506/contrib).

**Comment by joeyates** [comment]
Hi @michaldejmek 

I've implemented one of your suggestions - "the possibility to start a backup using a command, not a json file".

I've just released version 14.0.0, with the new `single backup` which let's you backup single accounts **without** a configuration file.
